runners/portability/java/src/main/java/org/apache/beam/runners/portability/PortableMetrics.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.portability;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.beam.model.jobmanagement.v1.JobApi;
+import org.apache.beam.model.pipeline.v1.MetricsApi;
+import org.apache.beam.sdk.metrics.DistributionResult;
+import org.apache.beam.sdk.metrics.GaugeResult;
+import org.apache.beam.sdk.metrics.MetricFiltering;
+import org.apache.beam.sdk.metrics.MetricKey;
+import org.apache.beam.sdk.metrics.MetricName;
+import org.apache.beam.sdk.metrics.MetricQueryResults;
+import org.apache.beam.sdk.metrics.MetricResult;
+import org.apache.beam.sdk.metrics.MetricResults;
+import org.apache.beam.sdk.metrics.MetricsFilter;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterables;
+import org.joda.time.Instant;
+
+public class PortableMetrics extends MetricResults {
+
+  private static final List<String> COUNTER_METRIC_TYPES =
+      ImmutableList.of("beam:metrics:sum_int_64");
+  private static final List<String> DISTRIBUTION_METRIC_TYPES =
+      ImmutableList.of("beam:metrics:distribution_int_64");
+  private static final List<String> GAUGE_METRIC_TYPES =
+      ImmutableList.of("beam:metrics:latest_int_64");
+
+  private static final String NAMESPACE_LABEL = "NAMESPACE";
+  private static final String METRIC_NAME_LABEL = "NAME";
+  private static final String STEP_NAME_LABEL = "PTRANSFORM";
+  private Iterable<MetricResult<Long>> counters;
+  private Iterable<MetricResult<DistributionResult>> distributions;
+  private Iterable<MetricResult<GaugeResult>> gauges;
+
+  private PortableMetrics(
```

Comment:
@mwalenia to answer your question, I'm indeed the correct person for MetricsPusher related questions.
Regarding MetricsPusher, the problem goes beyond the test itself. The whole MetricsPusher feature reports for now only user metrics (that is why the test sink that is tailored for it only reads user metrics). But the aim since the beginning of the architectural design (pull vs push essentially) was to allow in the future to support system metrics. Here is the design I did at the time: https://s.apache.org/runner_independent_metrics_extraction.
Long story short, the good thing to do IMHO is to enhance MetricsPusher to support system metrics as well and, of course, update the test/sink.